File: gpt-3.5-ft-large/stage2/host_data_construct_if_present_clause.f90
#ifndef T1
!T1:host_data construct if_present clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        REAL(8),DIMENSION(LOOPCOUNT):: a, b, c
        INTEGER:: x,r
        LOGICAL,DIMENSION(1):: devtest
        devtest(1) = .TRUE.
        
        !$acc enter data copyin(devtest(1:1))
        
        SEEDDIM(1) = 1
#       ifdef AMPI
        SEEDDIM(2) = MPISIZE
#       else
        SEEDDIM(2) = 1
#       endif
        CALL RANDOM_SEED(PUT=devtest(1), SOURCE=devtest(1), SIZE=1)
        CALL RANDOM_NUMBER(a)
        IF (DATA_TEST_MODE) THEN
          b(1:LOOPCOUNT) = 1
        ELSE
          CALL RANDOM_NUMBER(b)
        END IF
        c = 0
        
        !$acc data create(c(1:LOOPCOUNT))
          DO r = 0, PRESENT_or_not_AMPF(1, devtest)
            !$acc parallel
            IF (devtest(1)) THEN
              !$acc loop
              DO x = 1, LOOPCOUNT
                c(x) = c(x) + a(x) + b(x)
              END DO
            END IF
            !$acc end parallel
          END DO
          !$acc host_data if_present(c(1:LOOPCOUNT))
            DO x = 1, LOOPCOUNT
              IF (abs(c(x) - (a(x) + b(x))) .gt. PRECISION) THEN
                errors = errors + 1
              END IF
            END DO
          !$acc end host_data
        !$acc end data
        
        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM

File: Codellama-34b-Instruct/template/host_data_construct_if_present_clause.c
#include "acc_testsuite.h"

#ifndef T1
//T1:host_data construct if_present clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);

    #pragma acc host_data if_present
    {
        int a = 1;
        int b = 2;
        int c = 3;
        int d = 4;
        int e = 5;
        int f = 6;
        int g = 7;
        int h = 8;
        int i = 9;
        int j = 10;
        int k = 11;
        int l = 12;
        int m = 13;
        int n = 14;
        int o = 15;
        int p = 16;
        int q = 17;
        int r = 18;
        int s = 19;
        int t = 20;
        int u = 21;
        int v = 22;
        int w = 23;
        int x = 24;
        int y = 25;
        int z = 26;
    }

    if (a != 1 || b != 2 || c != 3 || d != 4 || e != 5 || f != 6 || g != 7 || h != 8 || i != 9 || j != 10 || k != 11 || l != 12 || m != 13 || n != 14 || o != 15 || p != 16 || q != 17 || r != 18 || s != 19 || t != 20 || u != 21 || v != 22 || w != 23 || x != 24 || y != 25 || z != 26) {
        err = 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}